Backend/src/main/java/com/onbank/exceptions/advice/ExceptionControllerAdvice.java

```diff
@@ -3,24 +3,40 @@
 import com.onbank.exceptions.ExceptionResponse;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 
+import javax.validation.ConstraintViolationException;
 import java.sql.Timestamp;
+import java.util.ArrayList;
 import java.util.Date;
 
 
 @ControllerAdvice
 public class ExceptionControllerAdvice {
 
+    private void printError(Exception e){
+        System.err.print(new Timestamp(new Date().getTime()) + "  EXCEPTION: ");
+        e.printStackTrace();
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ExceptionResponse> handle(final MethodArgumentNotValidException exception) {
+        BindingResult result = exception.getBindingResult();
+        String errorMessage = result.getFieldError().getDefaultMessage();
+        ExceptionResponse exceptionResponse = new ExceptionResponse(HttpStatus.BAD_REQUEST.value(), errorMessage);
+        this.printError(exception);
+        return new ResponseEntity<>(exceptionResponse, HttpStatus.BAD_REQUEST);
```

Comment:
Bardzo mi się podoba co tutaj zrobiłeś :) Bo to jest handler do tych walidacyjnych błędów z DTO tak?

Reply:
Tak :)